wizard/WizardController.qml

```diff
@@ -0,0 +1,482 @@
+        }
+
+        // make sure temporary wallet files are deleted
+        console.log("Removing temporary wallet: " + wizardController.tmpWalletFilename)
+        oshelper.removeTemporaryWallet(wizardController.tmpWalletFilename)
+
+        // protecting wallet with password
+        m_wallet.setPassword(walletOptionsPassword);
+
+        // Store password in session to be able to use password protected functions (e.g show seed)
+        appWindow.walletPassword = walletOptionsPassword
+
+        // save to persistent settings
+        persistentSettings.language = wizardController.language_language
+        persistentSettings.locale   = wizardController.language_locale
+
+        persistentSettings.account_name = wizardController.walletOptionsName
+        persistentSettings.wallet_path = new_wallet_filename
+        persistentSettings.restore_height = (isNaN(walletOptionsRestoreHeight))? 0 : walletOptionsRestoreHeight
+
+        persistentSettings.allow_background_mining = false
+        persistentSettings.is_recovering = (wizardController.walletOptionsIsRecovering === undefined) ? false : wizardController.walletOptionsIsRecovering
+        persistentSettings.is_recovering_from_device = (wizardController.walletOptionsIsRecoveringFromDevice === undefined) ? false : wizardController.walletOptionsIsRecoveringFromDevice
+    }
+
+    function createWalletFromDevice() {
+        // TODO: create wallet in temporary filename and a) move it to the path specified by user after the final
+        // page submitted or b) delete it when program closed before reaching final page
+
+        // Always delete the wallet object before creating new - we could be stepping back from recovering wallet
+        if (typeof wizardController.m_wallet !== 'undefined') {
+            walletManager.closeWallet()
+            console.log("deleting wallet")
+        }
+
+        var tmp_wallet_filename = oshelper.temporaryFilename();
+        console.log("Creating temporary wallet", tmp_wallet_filename)
+        var nettype = persistentSettings.nettype;
+        var restoreHeight = wizardController.walletOptionsRestoreHeight;
+        var subaddressLookahead = wizardController.walletOptionsSubaddressLookahead;
+        var deviceName = wizardController.walletOptionsDeviceName;
+
+        var wallet = walletManager.createWalletFromDevice(tmp_wallet_filename, "", nettype, deviceName, restoreHeight, subaddressLookahead);
+
+        var success = wallet.status === Wallet.Status_Ok;
+        if (success) {
+            wizardController.m_wallet = wallet;
+            wizardController.walletOptionsIsRecoveringFromDevice = true;
+            wizardController.tmpWalletFilename = tmp_wallet_filename;
+            wizardController.walletOptionsRestoreHeight = wizardController.m_wallet.walletCreationHeight;
+        } else {
+            console.log(wallet.errorString)
+            walletErrorDialog.text = wallet.errorString;
```

Comment:
`walletErrorDialog` is undefined as far as I can see.

Reply:
:+1: